arangod/Agency/State.cpp

```diff
@@ -748,7 +778,7 @@ bool State::loadCollections(TRI_vocbase_t* vocbase,
       std::shared_ptr<Buffer<uint8_t>> buf = std::make_shared<Buffer<uint8_t>>();
       VPackSlice value = arangodb::velocypack::Slice::emptyObjectSlice();
       buf->append(value.startAs<char const>(), value.byteSize());
-      _log.push_back(log_t(index_t(0), term_t(0), buf, std::string()));
+      _log.emplace_back(log_t(index_t(0), term_t(0), buf, std::string()));
```

Comment:
This change does not achieve anything, since a copy is done anyway. If you want to avoid a copy you have to lose the `log_t` constructor call. Here is a test program:

```
#include <iostream>
#include <deque>

class C {
  int a;
 public:
  C(int b) : a(b) {
    std::cout << "CONS " << a << std::endl;
  }
  C(C const& other) {
    a = other.a;
    std::cout << "Kopiert!!!" << std::endl;
  }
  C(C&& other) {
    a = other.a;
    other.a = 0;
    std::cout << "Moved!!!" << std::endl;
  }
  ~C() {
    std::cout << "DEST " << a << std::endl;
  }
};

int main() {
  std::deque<C> d;
  d.emplace_back(C(12));
  d.emplace_back(13);
  std::cout << d.size() << std::endl;
}
```

Reply:
Let's add a comment here that we do not need to modify `_clientIdLookupTable`, since the log entry generated has an empty clientId. This improves understandability.